packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -852,29 +854,32 @@ class DropdownButton<T> extends StatefulWidget {
   ///
   /// If the [onChanged] callback is null or the list of items is null
   /// then the dropdown button will be disabled, i.e. its arrow will be
-  /// displayed in grey and it will not respond to input. A disabled button
-  /// will display the [disabledHint] widget if it is non-null. If
-  /// [disabledHint] is also null but [hint] is non-null, [hint] will instead
-  /// be displayed.
+  /// displayed in grey and it will not respond to input.
   final List<DropdownMenuItem<T>>? items;
 
   /// The value of the currently selected [DropdownMenuItem].
   ///
-  /// If [value] is null and [hint] is non-null, the [hint] widget is
-  /// displayed as a placeholder for the dropdown button's value.
+  /// If [value] is null, [hint] or [disabledHint] is non-null, depends on the
+  /// dropdown enabled or disabled status, the [hint] or [disabledHint] widget
+  /// is displayed as a placeholder for the dropdown button's value.
   final T? value;
 
   /// A placeholder widget that is displayed by the dropdown button.
   ///
-  /// If [value] is null, this widget is displayed as a placeholder for
-  /// the dropdown button's value. This widget is also displayed if the button
-  /// is disabled ([items] or [onChanged] is null) and [disabledHint] is null.
+  /// If [value] is null, the dropdown is enabled ([items] and [onChanged] are non-null),
+  /// this widget is displayed as a placeholder for the dropdown button's value.
+  ///
+  /// If [value] is null, the dropdown is disabled and [disabledHint] is null,
+  /// This widget is displayed.
+  ///
+  /// If [value] is null, the dropdown is disabled and [disabledHint] is non-null,
+  /// [disabledHint] widget is displayed.
```

Comment:
Also, there's probably no need to describe [disabledHint]'s behavior here, since the main use-case of this parameter is for the enabled dropdown. We only should describe when this will be used if the dropdown is disabled:
```suggestion
  /// If [value] is null and the dropdown is enabled ([items] and [onChanged] are non-null),
  /// this widget is displayed as a placeholder for the dropdown button's value.
  ///
  /// If [value] is null and the dropdown is disabled and [disabledHint] is null,
  /// this widget is used as the placeholder.
```